stripe/src/main/java/com/stripe/android/model/Card.java

```diff
@@ -334,171 +415,321 @@ public boolean validateExpYear() {
         return expYear != null && !DateUtils.hasYearPassed(expYear);
     }
 
-    private boolean isValidLuhnNumber(String number) {
-        boolean isOdd = true;
-        int sum = 0;
-
-        for (int index = number.length() - 1; index >= 0; index--) {
-            char c = number.charAt(index);
-            if (!Character.isDigit(c)) {
-                return false;
-            }
-            int digitInteger = Integer.parseInt("" + c);
-            isOdd = !isOdd;
-
-            if (isOdd) {
-                digitInteger *= 2;
-            }
-
-            if (digitInteger > 9) {
-                digitInteger -= 9;
-            }
-
-            sum += digitInteger;
-        }
-
-        return sum % 10 == 0;
-    }
-
-    private String normalizeCardNumber(String number) {
-        if (number == null) {
-            return null;
-        }
-        return number.trim().replaceAll("\\s+|-", "");
-    }
-
+    /**
+     * @return the {@link #number} of this card
+     */
     public String getNumber() {
         return number;
     }
 
+    /**
+     * Setter for the card number. Note that mutating the number of this card object
+     * invalidates the {@link #brand} and {@link #last4}.
+     *
+     * @param number the new {@link #number}
+     */
+    @Deprecated
     public void setNumber(String number) {
```

Comment:
I'm deprecating the `setX` methods because this is a model, not an object to mess around with in memory. But if you do set the number, you very clearly change the `last4` and possibly the `brand`